### PR TITLE
GRV-1348 twig refactor in generator

### PR DESCRIPTION
**Pull request by hairmare**
Contains some ``@todo`` annotations and the following refactoring so far:

* refactor ``DynamicBundleBundle.php.twig`` to extend ``Bundle.php.twig``
* decompose large method in ``ResourceGenerator``
* clean up some templates so they generate nicer code (ie. indented code)

Here is what I still want to take care of in this PR:
* [x] ensure that generated things always render stuff via templates

Some of the todo make lots of sense but are not in the scope of this PR/GRV so I'll just leave them in the code where I dropped them.

### CODE REVIEW

File: src/Graviton/GeneratorBundle/Command/GenerateDynamicBundleCommand.php [hunk 1]
```diff
@@ -34,6 +35,8 @@
  * executed in the same context as the previous generator tools, and also those
  * used the shell (backtick operator to get git name/email for example).
  *
+ * @todo use symfony/process instead of shell_exec and/or create a new Application in-situ
```
Comment (lapistano) [inline src/Graviton/GeneratorBundle/Command/GenerateDynamicBundleCommand.php]: interesting in that context: http://php-and-symfony.matthiasnoback.nl/2014/03/test-symfony2-commands-using-the-process-component-and-asynchronous-assertions/

File: src/Graviton/GeneratorBundle/Manipulator/BundleBundleManipulator.php [hunk 2]
```diff
@@ -41,6 +41,8 @@ public function __construct(GravitonBundleInterface $bundle)
      * @return boolean
      *
      * @throws \RuntimeException If bundle is already defined
+     *
+     * @todo numer magic has to go, maybe parse and regenerate?
```
Comment (lapistano) [inline src/Graviton/GeneratorBundle/Manipulator/BundleBundleManipulator.php]: *number

File: src/Graviton/GeneratorBundle/Resources/skeleton/model/Model.php.twig [hunk 3]
```diff
@@ -15,11 +15,10 @@ use Graviton\RestBundle\Model\DocumentModel;
 class {{ document }} extends DocumentModel
```
Comment (lapistano) [inline src/Graviton/GeneratorBundle/Resources/skeleton/model/Model.php.twig]: *swisscom.ch

Comment (lapistano) [inline src/Graviton/GeneratorBundle/Resources/skeleton/model/Model.php.twig]: we might want to indicate in the DocBlock, that the resulting file was generated.

File: src/Graviton/GeneratorBundle/Command/GenerateDynamicBundleCommand.php (hunk 1 above)
Comment (lapistano) [inline src/Graviton/GeneratorBundle/Command/GenerateDynamicBundleCommand.php]: @todo move the detailed description/Warning to »Resources/docs« in order to keep the inline docu lean.

File: src/Graviton/GeneratorBundle/Command/GenerateDynamicBundleCommand.php [hunk 4]
```diff
@@ -476,6 +480,7 @@ private function getDefinitionsFromMongoDb()
             }
 
             $thisFilename = tempnam(sys_get_temp_dir(), 'mongoBundle_');
+            // @todo use symfony tools to write this
```
Comment (lapistano) [inline src/Graviton/GeneratorBundle/Command/GenerateDynamicBundleCommand.php]: +1

File: src/Graviton/GeneratorBundle/Resources/skeleton/model/Model.php.twig (hunk 3 above)
Comment (hairmare) [inline src/Graviton/GeneratorBundle/Resources/skeleton/model/Model.php.twig]: I'm not sure on this one. We should either mark up all generated files or none.

Personally I don't think it's needed in this context, the files are all in one .gitignored location anyway.

I know that doctrine usually marks up generated docs but symfony doesn't. For now I've been following symfony conventions on this.

File: src/Graviton/GeneratorBundle/Manipulator/BundleBundleManipulator.php (hunk 2 above)
Comment (hairmare) [inline src/Graviton/GeneratorBundle/Manipulator/BundleBundleManipulator.php]: I'll probably remove this before we merge. The code it was criticising looks exactly like it's counterpart in SensioGeneratorBundle as it stands now.

Maybe if I have a really good Idea I might change this. Still need to ifnd one though.

File: src/Graviton/GeneratorBundle/Resources/skeleton/model/Model.php.twig (hunk 3 above)
Comment (lapistano) [inline src/Graviton/GeneratorBundle/Resources/skeleton/model/Model.php.twig]: k

File: src/Graviton/GeneratorBundle/Command/GenerateDynamicBundleCommand.php (hunk 1 above)
Comment (hairmare) [inline src/Graviton/GeneratorBundle/Command/GenerateDynamicBundleCommand.php]: done

**Comment by lapistano** [comment]
@hairmare what is the status here? mergeable?

**Comment by hairmare** [comment]
I was hoping to get #34 done before this but yeah, this is probably good to go.

**Pull request merged**